### PR TITLE
Fix download gofile.py

### PR DESCRIPTION
**Pull request by jazz1611**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

I'm not a Python expert but did some research and fixed the download error from gofile.io. Hope you will improve the code more optimally

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/gofile.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import hashlib
+import requests
+import json
```
Comment (bashonly) [inline yt_dlp/extractor/gofile.py:3]: not needed, we should be using the helper methods instead
```suggestion
```

File: yt_dlp/extractor/gofile.py [hunk 2]
```diff
@@ -57,23 +59,32 @@ def _real_initialize(self):
             self._TOKEN = token.value
             return
 
-        account_data = self._download_json(
-            'https://api.gofile.io/createAccount', None, note='Getting a new guest account')
+        print('Getting a new guest account')
+        get_account_data = requests.post('https://api.gofile.io/accounts', '{}')
+        account_data = get_account_data.json()
+        # account_data = self._download_json(get_account_data, None, note='Getting a new guest account')
```
Comment (bashonly) [inline yt_dlp/extractor/gofile.py:65]:
```suggestion
        account_data = self._download_json(
            'https://api.gofile.io/accounts', None, 'Getting a new guest account', data=b'{}')
```

File: yt_dlp/extractor/gofile.py [hunk 3]
```diff
@@ -57,23 +59,32 @@ def _real_initialize(self):
...
         self._TOKEN = account_data['data']['token']
         self._set_cookie('.gofile.io', 'accountToken', self._TOKEN)
 
     def _entries(self, file_id):
         query_params = {
-            'contentId': file_id,
-            'token': self._TOKEN,
+            # 'contentId': file_id,
+            # 'token': self._TOKEN,
```
Comment (bashonly) [inline yt_dlp/extractor/gofile.py:72]:
```suggestion
```

File: yt_dlp/extractor/gofile.py [hunk 4]
```diff
@@ -57,23 +59,32 @@ def _real_initialize(self):
...
             'wt': '4fd6sg89d7s6',  # From https://gofile.io/dist/js/alljs.js
         }
         password = self.get_param('videopassword')
         if password:
             query_params['password'] = hashlib.sha256(password.encode('utf-8')).hexdigest()
```
Comment (bashonly) [inline yt_dlp/extractor/gofile.py:77]: do you know if video passwords still work the same way? the pw-protected extractor test is dead

File: yt_dlp/extractor/gofile.py [hunk 5]
```diff
@@ -57,23 +59,32 @@ def _real_initialize(self):
...
+        get_files = self._download_webpage(
+            'https://api.gofile.io/contents/' + file_id,
+            None,
+            headers={
+                'authority': 'api.gofile.io',
+                'authorization': 'Bearer ' + self._TOKEN,
+                'referer': 'https://gofile.io/',
+                'origin': 'https://gofile.io'
+            }, query=query_params)
+        files = json.loads(get_files)
```
Comment (bashonly) [inline yt_dlp/extractor/gofile.py:87]:
```suggestion
        files = self._download_json(
            f'https://api.gofile.io/contents/{file_id}', file_id, 'Getting filelist',
            query=query_params, headers={'Authorization': f'Bearer {self._TOKEN}'})
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/gofile.py (hunk 4 above)
Comment (jazz1611) [reply yt_dlp/extractor/gofile.py:77]: Sorry I didn't have a chance to try it so I don't know for sure

**Review by jazz1611 — COMMENTED** [review]
I tried changing according to your optimizations and it works fine

File: yt_dlp/extractor/gofile.py (hunk 4 above)
Comment (jazz1611) [reply yt_dlp/extractor/gofile.py:69]: I just received the password protected gofile.io URL through my friend and confirmed that it still works fine. This video may not be legal in some cases but we can try it as a test

`"https://gofile.io/d/OwrbtC" --video-password "12345"`

**Review by jazz1611 — COMMENTED** [review]
pw-protected extractor still working

**Review by bashonly — APPROVED** [review]

**Pull request merged**